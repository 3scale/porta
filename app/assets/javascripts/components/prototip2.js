

var Bubble = Class.create();
Bubble.prototype = {
  initialize: function(el) {
    this.offsetX = -8;
    this.offsetY = 38;

    this.target = el;
    // this.text = text;
    this.makeBubble();
  },
 
  makeBubble: function() {
    var bubble = $("bubble_" + this.target.title);
    // var bubble = new Element('div', { 'class': 'bubble'});
    // var content = new Element('div', { 'class': 'content'});
    // content.innerHTML = this.text;

    // bubble.appendChild(content);
    this.bubble = bubble;
    // new Insertion.Bottom(body, bubble);
    // bubble.hide();
    
    
    
    // set position

    // listen!
    
    this.target.observe('mouseover', (
        function() { 

          var position = this.target.cumulativeOffset();

          this.bubble.style.top = position[1] + this.offsetX + "px";
          this.bubble.style.left = position[0] + this.target.getWidth() + this.offsetY + "px";
          
          this.bubble.show(); 
        }).bind(this));
        
    this.target.observe('mouseout', (
        function() { 
          this.bubble.hide(); 
        }).bind(this));
        
  }
  
  
}

var BubbleHelper = {
  set: function(){
    $$('.define').each(function(l){
      new Bubble(l);      
    });
}}
  //     function(event) { 
  // 
  //       if ((event.element().className == 'submit' )) {
  // 
  //         var form = event.element().up('form');
  //         var formData = form.serialize(true);
  //         var url = form.action;
  //         
  //         this.ajaxPost(formData, url);
  //         event.stop();
  //       }
  //     })
  //   )        
  //   
  // }
