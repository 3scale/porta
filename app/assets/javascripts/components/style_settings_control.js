// Functions dealing with provider side control of site colours. Probably to be deprecated with the
// development of CMS system

jQuery.noConflict();

var settings = function() {
    // var B = ["background_color", "text_color", "link_color", "sidebar_fill_color", "sidebar_border_color"];
    var A = {
        updateColors: function() {
            var E = jQuery("#settings_bg_colour").val();
            var G = jQuery("#settings_text_colour").val();
            var F = jQuery("#settings_link_colour").val();
            var C = jQuery("#settings_menu_bg_colour").val();
						var D = jQuery("#settings_menu_link_colour").val();
						var H = jQuery("#settings_content_bg_colour").val();
						var J = jQuery("#settings_plans_tab_bg_colour").val();
						var K = jQuery("#settings_content_border_colour").val();
						
            jQuery("#previewContent").css({"background-color": E, color: G});

						jQuery('#pBox span').css("color", G);
						jQuery('#pBox  span a').css("color", F);
						jQuery('#pLinks span.active').css("background-color", H);
						jQuery('#pLinks span').css("color", D);
						jQuery('#pBox, #pFooter').css("background-color", H);
						jQuery('#pBox, #pFooter').css("border-color", K);
						jQuery('#previewPlans table th.head').css("background-color", J);
						
            jQuery("#pMenu").css({
                "background-color": C
            });
        },
        synchColors: function(D) {
            var C = settings.colorPicker.linked;
            if (C && C.value && C.value != settings.colorPicker.color) {
                C.value = settings.colorPicker.color;
                jQuery(C).css({
                    "background-color": C.value,
                    color: settings.colorPicker.hsl[2] > 0.5 ? "#000": "#fff"
                });
                settings.updateColors()
            }
        },
        initDesignSettings: function() {

          jQuery("#linkDesign a").click(function() {
              jQuery("#designSettings").show();
							jQuery('#detailSettings').hide();
              jQuery("#linkDetails a").removeClass("active");
              jQuery("#linkDesign a").addClass("active");
              jQuery("#linkDetails").removeClass("active");
              jQuery("#linkDesign").addClass("active");
							
              return false
          });

          jQuery("#linkDetails a").click(function() {
              jQuery("#designSettings").hide();
							jQuery('#detailSettings').show();

              // jQuery("#backgrounds").hide();
              jQuery("#linkDetails a").addClass("active");
              jQuery("#linkDesign a").removeClass("active");
              jQuery("#linkDetails").addClass("active");
              jQuery("#linkDesign").removeClass("active");

              // if (jQuery("#colors").css("display") == "none") {
              //     jQuery("#title_picker").removeClass("active");
              //     jQuery("#title_background").removeClass("inactive")
              // } else {
              //     jQuery("#title_picker").addClass("active");
              //     jQuery("#title_background").addClass("inactive")
              // }
              // if (jQuery("#user_profile_default").val()) {
              //     jQuery("#user_profile_default").val(false)
              // }
              return false
          });

            jQuery("#title_picker a").click(function() {
                jQuery("#colors").toggle();
                jQuery("#backgrounds").hide();
                jQuery("#title_background").removeClass("active");
                if (jQuery("#colors").css("display") == "none") {
                    jQuery("#title_picker").removeClass("active");
                    jQuery("#title_background").removeClass("inactive")
                } else {
                    jQuery("#title_picker").addClass("active");
                    jQuery("#title_background").addClass("inactive")
                }
                if (jQuery("#user_profile_default").val()) {
                    jQuery("#user_profile_default").val(false)
                }
                return false
            });
            jQuery("#title_background a").click(function() {
                jQuery("#backgrounds").toggle();
                jQuery("#colors").hide();
                jQuery("#title_picker").removeClass("active");
                if (jQuery("#backgrounds").css("display") == "none") {
                    jQuery("#title_background").removeClass("active");
                    jQuery("#title_picker").removeClass("inactive")
                } else {
                    jQuery("#title_background").addClass("active");
                    jQuery("#title_picker").addClass("inactive")
                }
                return false
            });
            jQuery("#themes a").click(function() {
                var E = jQuery(this).attr("id");
                settings.setTheme(E.substring(5));
                return false
            });
            jQuery("#backgrounds a").click(function() {
                jQuery("#backgrounds a").removeClass("active");
                jQuery(this).addClass("active");
                var E = jQuery(this).attr("id");
                if (E == "current_background") {
                    settings.setBackgroundImage(_tmp.currentImage)
                } else {
                    if (E == "no_background") {
                        settings.setBackgroundImage("none")
                    }
                }
                return false
            });
            jQuery("#user_profile_background_tile").click(function() {
                var E = jQuery("#user_profile_background_tile:checked").val();
                jQuery("body").css({
                    "background-repeat": E ? "repeat": "no-repeat",
                    "background-attachment": E ? "scroll": "fixed"
                });
                jQuery("#user_profile_use_background_image").val("true")
            });
            settings.colorPicker = jQuery.farbtastic("#picker", settings.synchColors);
            jQuery("#picker").click(function() {
                jQuery(".colorwell-selected").focus()
            });
            var D;
            var C = (function() {
                var E = [8, 9, 13, 16, 17, 18, 37, 38, 39, 40, 45, 46, 86, 88, 90];
                for (var F = 48; F <= 57; F++) {
                    E.push(F)
                }
                for (var F = 65; F <= 70; F++) {
                    E.push(F)
                }
                for (var F = 96; F <= 105; F++) {
                    E.push(F)
                }
                return E
            })();
            jQuery(".colorwell").each(settings.synchColors).focus(function() {
                if (D) {
                    jQuery(D).removeClass("colorwell-selected")
                }
                settings.colorPicker.linked = this;
                settings.colorPicker.setColor(this.value);
                jQuery("#picker").show();
                jQuery(D = this).addClass("colorwell-selected")
            }).blur(function() {
                var E = jQuery(E);
                E.val(E.val().toUpperCase())
            }).keydown(function(E) {
                if (jQuery.inArray(E.keyCode, C) != -1) {
                    if (jQuery.inArray(E.keyCode, [86, 88, 90]) != -1) {
                        return (E.ctrlKey || E.metaKey)
                    } else {
                        if ((jQuery.inArray(E.keyCode, [8, 46]) != -1) || ((E.keyCode == 88) && E.shiftKey)) {
                            return (jQuery(this).val() != "#")
                        } else {
                            if (E.shiftKey) {
                                return (jQuery.inArray(E.keyCode, [65, 66, 67, 68, 69, 70]) != -1)
                            }
                        }
                    }
                    return true
                }
                return false
            }).keyup(function(E) {
                if (jQuery(this).val().indexOf("#") != 0) {
                    jQuery(this).val("#" + jQuery(this).val())
                }
                settings.colorPicker.setColor(this.value);
                var F = jQuery(this).val();
                if (!document.all || (F.length == 7 || F.length == 4)) {
                    jQuery(this).css({
                        "background-color": jQuery(this).val(),
                        color: settings.colorPicker.hsl[2] > 0.5 ? "#000000": "#ffffff"
                    });
                    settings.updateColors()
                }
            });
            jQuery("#design_customization form a").click(settings.updateColors)
        }
    };
    return A
} ();

jQuery(document).ready(function() {
		if($('settingsBox')) 
    	settings.initDesignSettings();
});
