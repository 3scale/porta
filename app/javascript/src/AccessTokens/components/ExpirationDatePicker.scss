@import '~@patternfly/patternfly/patternfly-addons.css';

.pf-c-calendar-month,
.pf-c-form-control-expiration {
  width: 50%;
}

.pf-c-form-control-expiration {
  margin-right: 1em;
}

button.pf-c-form__group-label-help {
  min-width: auto;
  width: auto;
}
