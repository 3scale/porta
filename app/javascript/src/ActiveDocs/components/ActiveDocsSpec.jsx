// @flow

import React from 'react'

import { createReactWrapper } from 'utilities/createReactWrapper'
import SwaggerUI from 'swagger-ui-react'

import 'swagger-ui-react/swagger-ui.css'

type ActiveDocsSpecProps = {
  url: string
}

const ActiveDocsSpec = ({ url }: ActiveDocsSpecProps) => <SwaggerUI url={url} />

const ActiveDocsSpecWrapper = (props: ActiveDocsSpecProps, id: string) => createReactWrapper(<ActiveDocsSpec {...props} />, id)

export { ActiveDocsSpec, ActiveDocsSpecWrapper }
