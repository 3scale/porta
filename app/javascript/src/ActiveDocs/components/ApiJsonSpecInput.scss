div.CodeMirror {
  height: 30rem;
  resize: none;
}

textarea#api_docs_service_body[aria-invalid="true"] + div.CodeMirror {
  // HACK: Alert icon for CodeMirror text editor
  background-image: url("data:image/svg+xml;charset=utf8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 512 512'%3E%3Cpath fill='%23c9190b' d='M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z'/%3E%3C/svg%3E");
  background-position: calc(100% - 0.5rem) 0.5rem;
  background-repeat: no-repeat;
  background-size: 1rem 1rem;
  border-bottom-color: var(--pf-c-form__helper-text--m-error--Color);
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  border-bottom-style: solid;
  border-bottom-width: 2px;
}
