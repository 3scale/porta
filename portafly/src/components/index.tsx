export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
