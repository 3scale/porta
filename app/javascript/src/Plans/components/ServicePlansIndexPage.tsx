/* eslint-disable react/jsx-props-no-spreading */
import {
  PageSection,
  PageSectionVariants,
  Text,
  TextContent
} from '@patternfly/react-core'

import { createReactWrapper } from 'utilities/createReactWrapper'
import { IndexPageMainSection } from 'Plans/components/IndexPageMainSection'

import type { Props as MainSectionProps } from 'Plans/components/IndexPageMainSection'
import type { FunctionComponent } from 'react'

type Props = MainSectionProps

const ServicePlansIndexPage: FunctionComponent<Props> = ({
  defaultPlanSelectProps,
  plansTableProps
}) => (
  <>
    <PageSection variant={PageSectionVariants.light}>
      <TextContent>
        <Text component="h1">Service plans</Text>
        <Text component="p">
          Service plans allow you to define grades of service for each of the services (APIs)
          available through your developer portal. The plans allow you to define pricing per
          service and features available.
        </Text>
      </TextContent>
    </PageSection>

    <IndexPageMainSection
      defaultPlanSelectProps={defaultPlanSelectProps}
      plansTableProps={plansTableProps}
    />
  </>
)

const ServicePlansIndexPageWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ServicePlansIndexPage {...props} />, containerId) }

export type { Props }
export { ServicePlansIndexPage, ServicePlansIndexPageWrapper }
