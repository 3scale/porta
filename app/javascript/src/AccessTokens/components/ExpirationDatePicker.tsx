import { useState, useMemo } from 'react'
import { Alert, CalendarMonth, FormGroup, FormSelect, FormSelectOption } from '@patternfly/react-core'

import { createReactWrapper } from 'utilities/createReactWrapper'

import type { FunctionComponent, FormEvent } from 'react'

import './ExpirationDatePicker.scss'

interface ExpirationItem {
  id: string;
  label: string;
  period: number; // In seconds
}

const collection: ExpirationItem[] = [
  { id: '7', label: '7 days', period: 7 },
  { id: '30', label: '30 days', period: 30 },
  { id: '60', label: '60 days', period: 60 },
  { id: '90', label: '90 days', period: 90 },
  { id: 'custom', label: 'Custom...', period: 0 },
  { id: 'no-exp', label: 'No expiration', period: 0 }
]

const dayMs = 60 * 60 * 24 * 1000

interface Props {
  id: string;
  label: string | null;
}

const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label }) => {
  const [selectedItem, setSelectedItem] = useState(collection[0])
  const [pickedDate, setPickedDate] = useState(new Date())
  const fieldName = `human_${id}`
  const fieldLabel = label ?? 'Expires in'

  const fieldDate = useMemo(() => {
    if (selectedItem.period === 0) return null

    return new Date(new Date().getTime() + selectedItem.period * dayMs)
  }, [selectedItem])

  const fieldHint = useMemo(() => {
    if (!fieldDate) return

    const date = new Date(fieldDate)
    date.setHours(0, 0, 0, 0)

    return `The token will expire on ${date.toLocaleDateString()}`
  }, [fieldDate])

  const dateValue = useMemo(() => {
    let value = ''

    if (fieldDate) {
      value = fieldDate.toISOString()
    } else if (selectedItem.id === 'custom' ) {
      value = pickedDate.toISOString()
    }

    return value
  }, [fieldDate, selectedItem, pickedDate])

  const handleOnChange = (_value: string, event: FormEvent<HTMLSelectElement>) => {
    const value = (event.target as HTMLSelectElement).value
    const selected = collection.find(i => i.id === value) ?? null

    if (selected === null) return

    setSelectedItem(selected)
    setPickedDate(new Date())
  }

  return (
    <>
      <FormGroup
        isRequired
        fieldId={fieldName}
        label={fieldLabel}
      >
        <FormSelect
          className="pf-c-form-control-expiration"
          id={fieldName}
          value={selectedItem.id}
          onChange={handleOnChange}
        >
          {collection.map((item: ExpirationItem) => {
            return (
              <FormSelectOption
                key={item.id}
                label={item.label}
                value={item.id}
              />
            )
          })}
        </FormSelect>
        <span className="pf-c-form-control-expiration-hint">{fieldHint}</span>
      </FormGroup>
      <input id={id} name={id} type="hidden" value={dateValue} />
      {selectedItem.id === 'custom' && (
        <CalendarMonth className="pf-u-mt-md" date={pickedDate} onChange={setPickedDate} />
      )}
      {dateValue === '' && (
        <>
          <br />
          <Alert title="Expiration is recommended" variant="warning">
            It is strongly recommended that you set an expiration date for your token to help keep your information
            secure
          </Alert>
        </>
      )}
    </>
  )
}

const ExpirationDatePickerWrapper = (props: Props, containerId: string): void => { createReactWrapper(<ExpirationDatePicker id={props.id} label={props.label} />, containerId) }

export type { ExpirationItem, Props }
export { ExpirationDatePicker, ExpirationDatePickerWrapper }
