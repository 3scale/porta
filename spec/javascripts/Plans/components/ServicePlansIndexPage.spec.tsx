import { shallow } from 'enzyme'

import { ServicePlansIndexPage } from 'Plans/components/ServicePlansIndexPage'

import type { Props } from 'Plans/components/ServicePlansIndexPage'

const defaultProps = {
  defaultPlanSelectProps: {
    plans: [],
    initialDefaultPlan: null,
    path: '/path'
  },
  plansTableProps: {
    createButton: undefined,
    columns: [],
    plans: [],
    count: 0,
    searchHref: '/search'
  }
}

const shallowWrapper = (props: Partial<Props> = {}) => shallow(<ServicePlansIndexPage {...{ ...defaultProps, ...props }} />)

it('should render properly', () => {
  const wrapper = shallowWrapper()
  expect(wrapper).toMatchSnapshot()
})
