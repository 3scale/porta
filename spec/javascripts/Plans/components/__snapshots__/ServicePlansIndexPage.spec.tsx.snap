// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<Fragment>
  <PageSection
    variant="light"
  >
    <TextContent>
      <Text
        component="h1"
      >
        Service plans
      </Text>
      <Text
        component="p"
      >
        Service plans allow you to define grades of service for each of the services (APIs) available through your developer portal. The plans allow you to define pricing per service and features available.
      </Text>
    </TextContent>
  </PageSection>
  <IndexPageMainSection
    defaultPlanSelectProps={
      {
        "initialDefaultPlan": null,
        "path": "/path",
        "plans": [],
      }
    }
    plansTableProps={
      {
        "columns": [],
        "count": 0,
        "createButton": undefined,
        "plans": [],
        "searchHref": "/search",
      }
    }
  />
</Fragment>
`;
